src/compose/ComposeBox.js

```diff
@@ -217,6 +220,47 @@ class ComposeBox extends PureComponent<Props, State> {
     }
   };
 
+  insertAttachment = async (attachments: DocumentPickerResponse[]) => {
+    this.setState(({ numUploading }) => ({
+      numUploading: numUploading + 1,
+    }));
+
+    const { _, auth } = this.props;
+
+    let placeholders = '';
+    for (let i = 0; i < attachments.length; i++) {
+      const fileName = attachments[i].name ?? `attachment-${i + 1}`;
+      placeholders += `${_('[Uploading {fileName}...]()', { fileName })}\n\n`;
+    }
+    this.insertMessageTextAtCursorPosition(placeholders);
+
+    for (let i = 0; i < attachments.length; i++) {
+      const fileName = attachments[i].name ?? `attachment ${i + 1}`;
+      const placeholder = _('[Uploading {fileName}...]()', { fileName });
+      try {
+        const response = await api.uploadFile(auth, attachments[i].uri, fileName);
+        const linkText = `[${fileName}](${response.uri})`;
+        if (this.state.message.indexOf(placeholder) !== -1) {
+          this.setMessageInputValue(this.state.message.replace(placeholder, linkText));
+        } else {
+          this.setMessageInputValue(`${this.state.message} ${linkText}`);
```

Comment:
Perhaps a newline instead of a space? That seems like probably a cleaner result.

Reply:
Done.